pkg/types/baremetal/platform.go

```diff
@@ -21,6 +21,7 @@ type Host struct {
 	BootMACAddress  string                    `json:"bootMACAddress" validate:"required,uniqueField"`
 	HardwareProfile string                    `json:"hardwareProfile"`
 	RootDeviceHints *v1alpha1.RootDeviceHints `json:"rootDeviceHints,omitempty"`
+	BootMode        v1alpha1.BootMode         `json:"bootMode,omitempty"`
```

Comment:
This conflicts with https://github.com/openshift/installer/blob/master/docs/dev/explain.md#additional-guidelines-on-godoc-comments (1)

Embedding external types into install config API ties these 2 different API together in a way that could mean we cannot control our lifecycle ... Therefore all types in install config pick part of external API they want to implement and convert to the external API when necessary so that we can control our API at our pace.

Reply:
How should I address that in this case?

Reply:
Perhaps, we should redefine the RootDeviceHints and BootMode types locally in the platform and then have some helper function to copy data between here and BMH?

Reply:
That constraint is unfortunate. Since the installer doesn't use machine-api, but we want to keep the same interfaces for specifying hardware information, we'll have to have a lot of duplication.